packages/marqus-desktop/src/renderer/components/Monaco.tsx

```diff
@@ -49,14 +41,28 @@ export function Monaco(props: MonacoProps): JSX.Element {
   // if any of them change.
   const containerElement = useRef<HTMLDivElement | null>(null);
   const monacoEditor = useRef<monaco.editor.IStandaloneCodeEditor | null>(null);
-  const onChangeSub = useRef<monaco.IDisposable | null>(null);
+  const onChangeSub = useRef<monaco.IDisposable[]>([]);
   const activeNoteId = useRef<string | null>(null);
 
+  // Need to sub before doing anything or else we get no listener error.
+  useEffect(() => {
+    store.on("editor.boldSelectedText", boldSelectedText);
+    store.on("editor.italicSelectedText", italicSelectedText);
+    store.on("editor.setModelViewState", setModelViewState);
+
+    return () => {
+      store.off("editor.boldSelectedText", boldSelectedText);
+      store.off("editor.italicSelectedText", italicSelectedText);
+      store.off("editor.setModelViewState", setModelViewState);
+    };
+  }, [store]);
```

Comment:
This breaks our pattern of subscribing to events at the bottom of the component but I'm not sure of a better way forward.

Depending on the situation we may trigger `setModelViewState` on the first render so if we set up the listener at the bottom of the component the store throws an error due to not finding the listener.

Maybe we just leave as-is for now? I don't think it's worth changing the pattern yet.